store/storetest/job_store.go

```diff
@@ -360,24 +335,21 @@ func testJobUpdateOptimistically(t *testing.T, ss store.Store) {
 		"Foo": "Bar",
 	}
 
-	if updated, err2 := ss.Job().UpdateOptimistically(job, model.JOB_STATUS_SUCCESS); err2 != nil {
-		if updated {
-			t.Fatal("should have failed due to incorrect old status")
-		}
+	updated, err2 := ss.Job().UpdateOptimistically(job, model.JOB_STATUS_SUCCESS)
+	if updated {
+		require.Nil(t, err2,"should have failed due to incorrect old status")
 	}
 
 	time.Sleep(2 * time.Millisecond)
 
-	updated, err := ss.Job().UpdateOptimistically(job, model.JOB_STATUS_PENDING)
+	updated2, err := ss.Job().UpdateOptimistically(job, model.JOB_STATUS_PENDING)
 	require.Nil(t, err)
-	require.True(t, updated)
+	require.True(t, updated2)
 
 	updatedJob, err := ss.Job().Get(job.Id)
 	require.Nil(t, err)
 
-	if updatedJob.Type != job.Type || updatedJob.CreateAt != job.CreateAt || updatedJob.Status != job.Status || updatedJob.LastActivityAt <= job.LastActivityAt || updatedJob.Progress != job.Progress || updatedJob.Data["Foo"] != job.Data["Foo"] {
-		t.Fatal("Some update property was not as expected")
-	}
+	require.False(t, updatedJob.Type != job.Type || updatedJob.CreateAt != job.CreateAt || updatedJob.Status != job.Status || updatedJob.LastActivityAt <= job.LastActivityAt || updatedJob.Progress != job.Progress || updatedJob.Data["Foo"] != job.Data["Foo"], "Some update property was not as expected")
```

Comment:
This can be splitted in different `require.Equal` calls, one per each or in the `if`